rllib/agents/a3c/a3c.py

```diff
@@ -92,3 +110,60 @@ def execution_plan(
         train_op = grads.for_each(ApplyGradients(workers, update_all=False))
 
         return StandardMetricsReporting(train_op, workers, config)
+
+    def training_iteration(self) -> ResultDict:
+
+        # Shortcut.
+        local_worker = self.workers.local_worker()
+
+        def sample_and_compute_grads(worker: RolloutWorker):
+            """Call sample() and compute_gradients() remotely on workers."""
+            samples = worker.sample()
+            grads, infos = worker.compute_gradients(samples)
+            return {
+                "grads": grads,
+                "infos": infos,
+                "agent_steps": samples.agent_steps(),
+                "env_steps": samples.env_steps(),
+            }
+
+        # Perform rollouts and gradient calculations asynchronously.
+        with self._timers[GRAD_WAIT_TIMER]:
+            results = asynchronous_parallel_requests(
+                remote_requests_in_flight=self.remote_requests_in_flight,
+                actors=self.workers.remote_workers(),
+                ray_wait_timeout_s=0.0,
+                max_remote_requests_in_flight_per_actor=2,
+                remote_fn=sample_and_compute_grads,
+            )
+
+        global_vars = None
+        # Loop through all fetched worker-computed gradients (if any)
+        # and apply them - one by one - to the local worker's model.
+
+        result = None  # TODO
+        for worker, result in results.items():
+            # Apply gradients to local worker.
+            with self._timers[APPLY_GRADS_TIMER]:
+                local_worker.apply_gradients(result["grads"])
+            self._timers[APPLY_GRADS_TIMER].push_units_processed(result["agent_steps"])
+
+            # Update step counters.
+            self._counters[NUM_AGENT_STEPS_SAMPLED] += result["agent_steps"]
+            self._counters[NUM_ENV_STEPS_SAMPLED] += result["env_steps"]
+            # Create current global vars.
+            global_vars = {
+                "timestep": self._counters[NUM_AGENT_STEPS_SAMPLED],
+            }
+
+            # Synch updated weights back to the particular worker.
+            with self._timers[SYNCH_WORKER_WEIGHTS_TIMER]:
+                weights = local_worker.get_weights(local_worker.get_policies_to_train())
+                worker.set_weights.remote(weights, global_vars)
+
+        # Update global vars of the local worker.
+        if global_vars:
+            local_worker.set_global_vars(global_vars)
+
+        # TODO: If we have processed more than one gradients
```

Comment:
so to be clear we haven't written to result in this pr, right? But we want to for logging purposes.

Reply:
This is still WIP. I need to add proper compilation of the results dict. The only thing that's missing is to combine those learner stats from all workers that have returned something from the `async_parallel_requests` call further above. This shim implementation right now only returns the last one.
Let me finish this before merging, of course.